### PR TITLE
Trash modifications node

### PR DESCRIPTION
**Pull request by souissimai**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 1]
```diff
@@ -122,25 +122,53 @@ public List<ModificationInfos> getModifications(UUID groupUuid, boolean onlyMeta
...
+                .collect(Collectors.toList());
     }
 
     @Transactional(readOnly = true)
     public ModificationInfos getModificationInfo(UUID modificationUuid) {
         return modificationRepository
-            .findById(modificationUuid)
-            .orElseThrow(() -> new NetworkModificationException(MODIFICATION_NOT_FOUND, modificationUuid.toString()))
-            .toModificationInfos();
+                .findById(modificationUuid)
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:169]: Revert

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java [hunk 2]
```diff
@@ -200,11 +228,33 @@ public List<ModificationEntity> copyModificationsEntities(@NonNull UUID groupUui
...
+        }
+    }
+
     @Transactional
     public void updateModification(@NonNull UUID modificationUuid, @NonNull ModificationInfos modificationInfos) {
         this.modificationRepository
-            .findById(modificationUuid)
-            .orElseThrow(() -> new NetworkModificationException(MODIFICATION_NOT_FOUND, String.format("Modification (%s) not found", modificationUuid)))
-            .update(modificationInfos);
+                .findById(modificationUuid)
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java:256]: revert

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 3]
```diff
@@ -56,6 +56,15 @@ public ResponseEntity<List<ModificationInfos>> getNetworkModifications(@Paramete
         return ResponseEntity.ok().body(networkModificationService.getNetworkModifications(groupUuid, onlyMetadata, errorOnGroupNotFound));
     }
 
+    @GetMapping(value = "/groups/{groupUuid}/modifications-restore", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:59]: You could use the endpoint just above. And add a parameter 'stashedModification'

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 4]
```diff
@@ -201,6 +210,26 @@ public ResponseEntity<UUID> createModificationInGroup(@RequestBody ModificationI
         return ResponseEntity.ok().body(networkModificationService.createModificationInGroup(modificationsInfos));
     }
 
+    @PutMapping(value = "/network-modifications", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:213]: @PutMapping(value = "/network-modifications/trash" ?
A PUT request on /network-modifications should be a simple update

Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:213]: @PutMapping(value = "/network-modifications/stash" ?
A PUT request on /network-modifications should be a simple update

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java [hunk 5]
```diff
@@ -201,6 +210,26 @@ public ResponseEntity<UUID> createModificationInGroup(@RequestBody ModificationI
...
+    @Operation(summary = "put network modifications into trash")
+    @ApiResponse(responseCode = "200", description = "The network modifications were deleted")
+    public ResponseEntity<Void> putNetworkModificationsIntoTrash(
+            @Parameter(description = "Network modification UUIDs") @RequestParam("uuids") List<UUID> networkModificationUuids,
+            @Parameter(description = "Group UUID") @RequestParam("groupUuid") UUID groupUuid) {
+        networkModificationService.putNetworkModificationsIntoTrash(networkModificationUuids);
+        return ResponseEntity.ok().build();
+    }
+
+    @PutMapping(value = "/restore-network-modifications", produces = MediaType.APPLICATION_JSON_VALUE)
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/NetworkModificationController.java:223]: @PutMapping(value = "/network-modifications/restore" ?

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java [hunk 6]
```diff
@@ -40,6 +40,7 @@ public ModificationGroupEntity(UUID uuid) {
 
     public void addModification(ModificationEntity modification) {
         modifications.add(modification);
+        modification.setIsRestored(true);
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java:43]: modification.setStashed(false);

File: src/main/java/org/gridsuite/modification/server/dto/ModificationInfos.java [hunk 7]
```diff
@@ -78,6 +78,9 @@ public class ModificationInfos {
     @Schema(description = "Modification date")
     private ZonedDateTime date;
 
+    @Schema(description = "Modification flag")
```
[the remark this answers is not in the document]
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/dto/ModificationInfos.java:81]: done

File: src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java (hunk 1 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/repositories/NetworkModificationRepository.java]: done

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java (hunk 3 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: parameter added

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java (hunk 4 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: changed

File: src/main/java/org/gridsuite/modification/server/NetworkModificationController.java (hunk 5 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/NetworkModificationController.java]: done

File: src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java (hunk 6 above)
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/entities/ModificationGroupEntity.java]: done

File: src/main/resources/db/changelog/changesets/changelog_20230831T125009Z.xml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog" xmlns:ext="http://www.liquibase.org/xml/ns/dbchangelog-ext" xmlns:pro="http://www.liquibase.org/xml/ns/pro" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog-ext http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-ext.xsd http://www.liquibase.org/xml/ns/pro http://www.liquibase.org/xml/ns/pro/liquibase-pro-4.1.xsd http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.1.xsd">
+    <changeSet author="souissimai (generated)" id="1693486224304-6">
+        <addColumn tableName="modification">
+            <column name="stashed" type="boolean"/>
```
Comment (etiennehomer) [inline src/main/resources/db/changelog/changesets/changelog_20230831T125009Z.xml:5]: you should set a default value for old modifications
Something like : defaultValue=false

Comment (souissimai) [reply src/main/resources/db/changelog/changesets/changelog_20230831T125009Z.xml]: done

File: src/main/java/org/gridsuite/modification/server/dto/ModificationInfos.java [hunk 9]
```diff
@@ -78,6 +78,9 @@ public class ModificationInfos {
...
+    private Boolean stashed;
```
Comment (etiennehomer) [inline src/main/java/org/gridsuite/modification/server/dto/ModificationInfos.java:82]: isStashed would be better as it's a boolean

**Review by etiennehomer — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java [hunk 10]
```diff
@@ -102,6 +108,16 @@ public void updateNetworkModification(@NonNull UUID modificationUuid, @NonNull M
         networkModificationRepository.updateModification(modificationUuid, modificationInfos);
     }
 
+    @Transactional
+    public void putNetworkModificationsIntoTrash(@NonNull List<UUID> modificationUuids) {
```
[the remark this answers is not in the document]
Comment (souissimai) [reply src/main/java/org/gridsuite/modification/server/service/NetworkModificationService.java]: done

**Review by ayolab — APPROVED** [review]

**Pull request merged**